repotracker/repotracker.go

```diff
@@ -520,15 +520,16 @@ func makeBuildBreakSubscriber(userID string) (*event.Subscriber, error) {
 	return subscriber, nil
 }
 
-func CreateManifest(v model.Version, proj *model.Project, branch string, settings *evergreen.Settings) (*manifest.Manifest, error) {
+func CreateManifest(v model.Version, proj *model.Project, projectRef *model.ProjectRef, settings *evergreen.Settings) (*manifest.Manifest, error) {
 	if len(proj.Modules) == 0 {
 		return nil, nil
 	}
 	newManifest := &manifest.Manifest{
 		Id:          v.Id,
 		Revision:    v.Revision,
 		ProjectName: v.Identifier,
-		Branch:      branch,
+		Branch:      projectRef.Branch,
+		IsBase:      v.Requester == evergreen.RepotrackerVersionRequester,
```

Comment:
how does backwards compatibility for this work for the data in the database?

Reply:
if a base commit was created before this is deployed we fall back to it in `FindFromVersion` (we look first for a commit that is marked as the base commit and if we don't find one we look at any manifest for the same project and revision).

Maybe an alternative makes more sense: manifests for patches and triggers are marked `NotBase` (and the query would look for `not_base: {$ne: true}` to find the base commit's manifest).

Reply:
I switched to the `NotBase` strategy

Reply:
I don't think this address it completely since manifests created before this is deployed will always be returned by ByBaseProjectAndRevision. This might need an actual migration to set it for patch manifests

Reply:
I agree that it will not fix the issue for old base commits without a migration.
I'm open to doing a migration if we deem it necessary. (In that case I'd probably switch back to `IsBase`). Is `anser` the answer? Does it work (with the Go driver)?

Reply:
(pardon the pun)

Reply:
Anser isn't working yet, so the approach has been to write a one-off migration with a mongo shell script. I think the logic is simple enough here that it wouldn't be hard to do for this

Reply:
Okay. I reverted the last commit.
I think the plan should be to:
1) deploy these changes,
2) run a migration,
3) remove the check for a legacy manifest